### PR TITLE
Solve `2023-03-29` nightly build failure issues

### PR DESCRIPTION
**Pull request by bendnorman**
# PR Overview

We got a few failures in the most recent nightly build. It looks like there are [two new PRs](https://github.com/catalyst-cooperative/pudl/pulls?q=is%3Apr+is%3Aclosed+merged%3A2023-03-28..2023-03-29+base%3Adev+sort%3Aupdated-desc+) that made it into the build:
1. A log file for the build is not being written to the output bucket or sent to the #pudl-deployments slack channel. I pulled the test errors below from [Cloud Logging](https://console.cloud.google.com/logs/query;query=resource.type%3D%22gce_instance%22%20resource.labels.instance_id%3D%222178497257830677871%22;timeRange=2023-03-29T09:12:49.926Z%2F2023-03-29T16:12:49.926935Z;cursorTimestamp=2023-03-29T14:15:38.299009740Z?project=catalyst-cooperative-pudl). 

2. `test/validate/eia_test.py::test_no_null_cols_eia` failed on the `frc_eia923` table:
```
E           ValueError: Null columns found in frc_eia923: ['bulk_agg_fuel_cost_per_mmbtu']
```
@zaneselvans @e-belfer did #2447 touch this table?

3. `test/integrations/etl_test.py:: test_pudl_engine` found a foreign key constraint failure:
```
E           pudl.io_managers.ForeignKeyErrors: Foreign key error for table: purchased_power_ferc1 -- power_purchase_types_ferc1 {self.foreign_key} -- on rows {self.rowids}
```
I'm a dingdong and forgot to make the second half of the error message an f-string. `purchased_power_ferc1` is the child table and `power_purchase_types_ferc1` is the parent table. I don't think either of the two PRs directly changed anything about FERC tables. I'm going to download `pudl.sqlite` and run the fk checks locally to debug this.

# PR Checklist

- [ ] Merge the most recent version of the branch you are merging into (probably `dev`).
- [ ] All CI checks are passing. [Run tests locally to debug failures](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#running-tests-with-tox)
- [ ] Make sure you've included good docstrings.
- [ ] For major data coverage & analysis changes, [run data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation)
- [ ] Include unit tests for new functions and classes.
- [ ] Defensive data quality/sanity checks in analyses & data processing functions.
- [ ] Update the [release notes](https://catalystcoop-pudl.readthedocs.io/en/latest/release_notes.html) and reference reference the PR and related issues.
- [ ] Do your own explanatory review of the PR to help the reviewer understand what's going on and identify issues preemptively.

### CODE REVIEW

**Comment by zaneselvans** [comment]
This seems like a vote for increasing the priority of setting up the `check_foreign_keys()` follow-on job!

#2447 should not have touched any of this stuff.  It doesn't even create a new raw asset b/c of the weird archive problem with the 2018 data, so all it adds is some additional spreadsheet mapping metadata.

So far as i know nothing has changed with the ferc1 tables either so... that seems very odd!

**Comment by jdangerx** [comment]
I think we lost the logs because we don't have `pudl-etl.log` anymore:

image

Which happened when we stopped using `configure_root_logger` with a `log_file` set. I can push some changes to this branch to test out if pudl-etl.log gets written.

**Comment by bendnorman** [comment]
In the past I don't think the `pudl-etl.log` file was being created using a python logger. We were just piping the stdout and stderr into a new file called `pudl-etl.log`. I don't think it's advised to have multiple processing writing to [a logfile via a python logger](https://docs.python.org/3/howto/logging-cookbook.html#logging-to-a-single-file-from-multiple-processes).

I think the path for `$LOGFILE` might be incorrect.

File: docker/gcp_pudl_etl.sh [hunk 1]
```diff
@@ -24,25 +24,30 @@ function run_pudl_etl() {
...
+        --log-file=${PUDL_OUTPUT}/pytest.log \
         --live-dbs test
 }
 
 function shutdown_vm() {
     # Copy the outputs to the GCS bucket
     gsutil -m cp -r $PUDL_OUTPUT "gs://nightly-build-outputs.catalyst.coop/$ACTION_SHA-$GITHUB_REF"
 
-    upload_file_to_slack "${PUDL_OUTPUT}/pudl-etl.log" "Logs for $ACTION_SHA-$GITHUB_REF:"
+    upload_file_to_slack "${PUDL_OUTPUT}/ferc-to-sqlite.log" "ferc_to_sqlite logs for $ACTION_SHA-$GITHUB_REF:"
```
Comment (jdangerx) [inline docker/gcp_pudl_etl.sh:48]: Do we prefer looking at raw logs vs. in Cloud Logging? Seems like the extra filtering tools from Cloud Logging might be handy, and the fact that they stream in vs. being uploaded at the end is nice too.

**Comment by jdangerx** [comment]
> In the past I don't think the `pudl-etl.log` file was being created using a python logger. We were just piping the stdout and stderr into a new file called `pudl-etl.log`. I don't think it's advised to have multiple processing writing to [a logfile via a python logger](https://docs.python.org/3/howto/logging-cookbook.html#logging-to-a-single-file-from-multiple-processes).
>  ...
> I think all we have to do is update the `$LOGFILE` path.

Ahh, yeah makes sense. I'll do that then, and leave the "should we just be using Cloud Logging" question as a non-blocking discussion :)

**Comment by bendnorman** [comment]
The `frc_eia923` error failed for the same reason the FK constraint test failed. The `fuel_receipts_costs_aggs_eia` table was not loaded to the db because of [a `MetadataDiffError`](https://console.cloud.google.com/logs/query;cursorTimestamp=2023-03-29T06:27:10.310851467Z;pinnedLogId=2023-03-29T06:27:11.268593387Z%2Fx4cz66svvse2rik25;query=resource.type%3D%22gce_instance%22%20resource.labels.instance_id%3D%222178497257830677871%22%0A%0A%22fuel_receipts_costs_aggs_eia%22%0Atimestamp%3D%222023-03-29T06:27:10.310851467Z%22%0AinsertId%3D%22x4cz66svvse2rij25%22;resultsSearch=power_purchase_types_ferc1;timeRange=2023-03-29T06:12:49.926Z%2F2023-03-30T02:48:55.556Z?project=catalyst-cooperative-pudl).

**Comment by jdangerx** [comment]
I agree that we should revert #2331 and re-evaluate our approach - then at least we won't be blocking all the other work that wants to get into `dev`.

**Comment by bendnorman** [comment]
Sounds good to me. Let's revert it, and I'll add some thoughts to #2377 .

**Comment by zaneselvans** [comment]
I attempted to merge this into `dev` before the nightly build kicked off but missed it by 1 minute.

**Comment by zaneselvans** [comment]
I cancelled the scheduled deployment since it seemed doomed, and kicked one off manually on the post merge dev.

**Pull request merged**